api-reference/_hidden/dxDataGridColumn/calculateGroupValue.md

```diff
@@ -4,16 +4,172 @@ type: String | function(rowData)
 ---
 ---
 ##### shortDescription
-Specifies a field name or a function that returns a field name or a value to be used for grouping column cells.
+A field name by which to group cells or a function that returns a field name or a value to perform this operation. This value is also used to sort **DataGrid**'s groups.
 
 ##### param(rowData): Object
 The current row's data.
 
 ##### return: any
-The calculated value to be used for grouping.
+The calculated value to be used for grouping and group sorting.
+
+---
+
+If you want to use modified values of a grouping column, assign
+
+By default, [grouping](/concepts/05%20Widgets/DataGrid/45%20Grouping '/Documentation/Guide/Widgets/DataGrid/Grouping/') is conducted using the exact values that the grouping column contains. If you want to use calculated values, implement the **calculateGroupValue** function. As a consequence, group cells' headers may display extra or missing information. To remove or add it, implement the [groupCellTemplate](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#groupCellTemplate) option ([groupCellRender](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#groupCellRender) in React). The example below displays the *"California"* group always on top regardless of the sort order. For this, **calculateGroupValue** and **groupCellTemplate** are used:
```

Comment:
>By default, [grouping](/concepts/05%20Widgets/DataGrid/45%20Grouping '/Documentation/Guide/Widgets/DataGrid/Grouping/') is conducted using the exact values that the grouping column contains.

We should rewrite this old sentence to avoid 'by default'. How about the following: 'The **DataGrid** groups columns by the exact values that the grouping column contains. 

>As a consequence, group cells' headers may display extra or missing information.

I think that headers cannot display **missing** information. That's why it is missing, isn't it?
I would also replace 'As a consequence' with something less complicated - maybe 'After that' or similar.
.
>To remove or add it, implement the [groupCellTemplate](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#groupCellTemplate) option

To display the information correctly, implement...

>For this, **calculateGroupValue** and **groupCellTemplate** are used

Let's get rid of passive. We may use the first person plural instead.